py/plugins/google-genai/src/genkit/plugins/google_genai/google.py

```diff
@@ -165,6 +169,124 @@ def initialize(self, ai: GenkitRegistry) -> None:
             ai.define_model(name=vertexai_name(version), fn=imagen_model.generate, metadata=imagen_model.metadata)
 
 
+class VertexAIVectorSearch(Plugin):
+    """VertexAI vector store plugin for Genkit."""
+
+    name: str = 'vertexAIVectorstore'
+
+    def __init__(
+        self,
+        retriever: Type[VertexAIVectorStoreRetriever],
```

Comment:
I feel `retriever` is too overloaded -- `VertexAIVectorSearch` defines a retriever... so in order to define a retriever you have to pass in a retriever.
In JS they are called `documentRetriever`... so maybe `doc_retriever`?